wikiclue/src/routes/daily/+page.svelte

```diff
@@ -0,0 +1,182 @@
+
+	const isOverlayOpen = writable(false);
+	let searchTerm = '';
+	const wordsToFind = ['Sample', 'Words']; // this will need to change to actual algo code
+	let incorrectAnswer = false;
+	let gameOver = false;
+	let guessesRemaining = 6;
+	let today = new Date();
+	let maxStreak = 0;
+	let currentStreak = 0;
+	let totalWins = 0;
+	let totalTimesPlayed = 0;
+	let guessDistribution: number[] = [];
+	let statistics = { Played: 0, 'Win %': '0%', 'Current Streak': 0, 'Max Streak': 0 };
+	let monthNames = [
+		'January',
+		'February',
+		'March',
+		'April',
+		'May',
+		'June',
+		'July',
+		'August',
+		'September',
+		'October',
+		'November',
+		'December'
+	];
+
+	onMount(() => {
+		loadDailyWords();
+		getUserInfo();
+	});
+
+	function loadDailyWords() {
+		// Load daily words from firebase
+		if (browser) {
+			const savedFirstWord = sessionStorage.getItem('firstWord');
+			const savedSecondWord = sessionStorage.getItem('secondWord');
+			if (savedFirstWord !== null && savedSecondWord !== null) {
+				wordsToFind[0] = JSON.parse(savedFirstWord);
+				wordsToFind[1] = JSON.parse(savedSecondWord);
+			}
+		}
+	}
+
+	function getUserInfo() {
+		// Get user info from firebase
+		let lastDate = new Date(); // last date completed daily challenge
+		let savedRemaining = 2; // guesses remaining on daily challenge
+		maxStreak = 7;
+		currentStreak = 5;
+		totalWins = 30;
+		totalTimesPlayed = 50;
+		guessDistribution = [4, 1, 3, 3, 4, 1];
+		// all above is from api call
+
+		if (
+			lastDate.getFullYear() === today.getFullYear() &&
+			lastDate.getMonth() === today.getMonth() &&
+			lastDate.getDay() === today.getDay()
+		) {
+			if (savedRemaining === 0) {
+				endGame();
+			}
+			guessesRemaining = savedRemaining;
+		} else {
+			totalTimesPlayed++;
+			storeData();
+		}
+	}
+
+	function confirmPressed() {
+		incorrectAnswer = false;
+		// will need to change the if statement to use actual wikipedia api function
+		if (searchTerm.includes(wordsToFind[0]) && searchTerm.includes(wordsToFind[1])) {
+			currentStreak++;
+			totalWins++;
+			guessDistribution[6 - guessesRemaining]++;
+			if (currentStreak > maxStreak) maxStreak = currentStreak;
+			guessesRemaining = 0;
+			storeData();
+			endGame();
+		} else if (guessesRemaining <= 1) {
+			guessesRemaining = 0;
+			currentStreak = 0;
+			storeData();
+			endGame();
+		} else {
+			incorrectAnswer = true;
+			guessesRemaining--;
+			setTimeout(() => {
+				incorrectAnswer = false;
+			}, 2000);
+			storeData();
+		}
+	}
+
+	function storeData() {
+		// save into database
+		// lastDate = today
+		// guesses_remaing = guessesRemaining
+		// max_streak = maxStreak
+		// current_streak = currentStreak
+		// total_wins = totalWins
+		// total_times_played = totalTimesPlayed
+		// guesses_distribution = guessesDistribution
+	}
+
+	function formatOverlay() {
+		const winPercentage = ((totalWins / totalTimesPlayed) * 100).toFixed(0);
+		statistics = {
```

Comment:
I feel like the string formatting here should be in enddaily instead. That way u could just construct the raw data in formatOverlay

Reply:
I want to discuss this one

Reply:
Fair, we need to discuss the end overlays anyways. If u look at the other PR thats open for endlevels, I didn't even use the endlevel component because it was causing me grief. We can talk about it in the sync up